packages/react-core/src/components/Menu/MenuItem.tsx

```diff
@@ -70,6 +66,22 @@ export interface MenuItemProps extends Omit<React.HTMLProps<HTMLLIElement>, 'onC
   innerRef?: React.Ref<HTMLAnchorElement | HTMLButtonElement>;
 }
 
+type ConditionalMenuItemProps =
+  | {
+      /** Target navigation link */
+      to?: string;
+      /** @beta Flyout menu. Disallowed if nav link is defined */
+      flyoutMenu?: never;
+    }
+  | {
+      /** Target navigation link. Disallowed if flyoutMenu is defined */
+      to?: never;
+      /** @beta Flyout menu */
+      flyoutMenu?: React.ReactElement;
+    };
```

Comment:
@tlabaj Would this change be breaking? I know it's an a11y bug, but I feel like there's a decent chance that there are consumers who followed the example and would need to make changes for the proposed type change here. Also I know that we have the beta tag on `flyoutMenu`, but we don't have it on `to`.

Genuine question as I'm not sure. A conditional type like this is a really neat idea though.

Reply:
Another thing that was noticed in our React huddle is that by doing this the to and flyoutMenu props are no longer rendered in the props table on the React page for the component. If anything, perhaps we can put the implementation proposed here on the backburner and discuss it further down the line?

Reply:
yes.  This would be a breaking change.
I would probably not recommend defining type this way and instead have 2 separate types and a prop that could be one or the other.
We would need to wait for the breaking change release to do this.